apps/app/src/components/Admin/UserGroupDetail/UserGroupUserFormByInput.tsx

```diff
@@ -0,0 +1,127 @@
+import type { FC } from 'react';
+import React, { useState } from 'react';
+
+import type { IUserGroupHasId, IUserHasId } from '@growi/core';
+import { UserPicture } from '@growi/ui/dist/components';
+import { useTranslation } from 'next-i18next';
+import { AsyncTypeahead } from 'react-bootstrap-typeahead';
+import { debounce } from 'throttle-debounce';
+
+import { toastSuccess, toastError } from '~/client/util/toastr';
+import Xss from '~/services/xss';
+
+type Props = {
+  isAlsoMailSearched: boolean,
+  isAlsoNameSearched: boolean,
+  onClickAddUserBtn: (username: string) => Promise<void>,
+  onSearchApplicableUsers: (searchWord: string) => Promise<IUserHasId[]>,
+  userGroup: IUserGroupHasId,
+}
+
+export const UserGroupUserFormByInput: FC<Props> = (props) => {
+  const {
+    isAlsoMailSearched, isAlsoNameSearched, onClickAddUserBtn, onSearchApplicableUsers, userGroup,
+  } = props;
+
+  const { t } = useTranslation();
+  const [keyword, setKeyword] = useState('');
+  const [inputUser, setInputUser] = useState<IUserHasId[]>([]);
+  const [applicableUsers, setApplicableUsers] = useState<IUserHasId[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSearchError, setIsSearchError] = useState(false);
+
+  const xss = new Xss();
+
+  const addUserBySubmit = async() => {
+    if (inputUser.length === 0) { return }
+    const userName = inputUser[0].username;
+
+    try {
+      await onClickAddUserBtn(userName);
+      toastSuccess(`Added "${xss.process(userName)}" to "${xss.process(userGroup.name)}"`);
+      setInputUser([]);
+    }
+    catch (err) {
+      toastError(new Error(`Unable to add "${xss.process(userName)}" to "${xss.process(userGroup.name)}"`));
+    }
+  };
+
+  const searchApplicableUsers = async() => {
+    try {
+      const users = await onSearchApplicableUsers(keyword);
+      setApplicableUsers(users);
+      setIsLoading(false);
+    }
+    catch (err) {
+      setIsSearchError(true);
+      toastError(err);
+    }
+  };
+
+  const searchApplicableUsersDebounce = debounce(1000, searchApplicableUsers);
+
+  const handleChange = (inputUser: IUserHasId[]) => {
+    setInputUser(inputUser);
+  };
+
+  const handleSearch = (keyword) => {
+    if (keyword === '') {
+      return;
+    }
+
+    setKeyword(keyword);
+    setIsLoading(true);
+    searchApplicableUsersDebounce();
+  };
+
+  const onKeyDown = (event) => {
+    // 13 is Enter key
+    if (event.keyCode === 13) {
```

Comment:
event.keyCode は非推奨なので event.key を使うように変更お願いします。
https://developer.mozilla.org/ja/docs/Web/API/KeyboardEvent/keyCode

```suggestion
    if (event.key === 'Enter') {
```

できる限り引数に型の指定お願いします。上記も未然に防ぐことができます。

Reply:
修正と型を追加しました。